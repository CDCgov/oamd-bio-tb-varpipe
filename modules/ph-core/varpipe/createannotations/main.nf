process VARPIPE_CREATEANNOTATIONS {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(annotation)

    output:
    tuple val(meta), path("*_DR_loci_annotation.txt")      , optional:true, emit: drLociAnnotation
    tuple val(meta), path("*_full_annotation.txt")         , optional:true, emit: fullAnnotation
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'create_annotation.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_DR_loci_Final_annotation.txt
    touch ${prefix}_full_Final_annotation.txt
    touch ${prefix}_DR_loci_annotation.txt
    touch ${prefix}_full_annotation.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
