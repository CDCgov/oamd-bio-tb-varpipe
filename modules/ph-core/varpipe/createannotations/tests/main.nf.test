nextflow_process {

    name "Test Process VARPIPE_CREATEANNOTATIONS"
    script "../main.nf"
    process "VARPIPE_CREATEANNOTATIONS"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/createannotations"

    test("varpipe create DR loci annotation [vcf]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_raw_annotation.txt'),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe create full annotation [vcf]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_full_raw_annotation.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe create full annotation [vcf] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_full_raw_annotation.txt')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe create DR loci annotation [vcf] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_raw_annotation.txt'),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
