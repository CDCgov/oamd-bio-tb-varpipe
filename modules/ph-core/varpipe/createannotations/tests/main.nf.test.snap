{
    "varpipe create full annotation [vcf] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3d0dc3c159bb0a22258a6a95303dad7e"
                ],
                "drLociAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fullAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3d0dc3c159bb0a22258a6a95303dad7e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T16:32:05.80522583"
    },
    "varpipe create full annotation [vcf]": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_annotation.txt:md5,24c6026a46f7bc321d5a79762fe7e962"
                    ]
                ],
                "2": [
                    "versions.yml:md5,2202bc135fc36c20129980789889aed6"
                ],
                "drLociAnnotation": [
                    
                ],
                "fullAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_annotation.txt:md5,24c6026a46f7bc321d5a79762fe7e962"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2202bc135fc36c20129980789889aed6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T15:34:33.34694024"
    },
    "varpipe create DR loci annotation [vcf]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_annotation.txt:md5,00b8022f34c5bbaa93f59f22857223b9"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,2202bc135fc36c20129980789889aed6"
                ],
                "drLociAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_annotation.txt:md5,00b8022f34c5bbaa93f59f22857223b9"
                    ]
                ],
                "fullAnnotation": [
                    
                ],
                "versions": [
                    "versions.yml:md5,2202bc135fc36c20129980789889aed6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T15:35:38.496658182"
    },
    "varpipe create DR loci annotation [vcf] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3d0dc3c159bb0a22258a6a95303dad7e"
                ],
                "drLociAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_DR_loci_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fullAnnotation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_full_annotation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3d0dc3c159bb0a22258a6a95303dad7e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T16:32:12.734679216"
    }
}
